### ISSUE TITLE
black bar over Gujarat

### ISSUE THREAD

**Issue by hn-88** [OpenSpace/OpenSpace#3040]
Hi, this black bar is seen at an altitude of around 2000 km, vanishes when we go farther away or come nearer. So this must be some sort of artifact of this particular data layer, I suppose. Unfortunately, when I create a "zoom in" fulldome image sequence, this black bar also appears in the sequence. Any tips on how to get rid of it? I tried moving the date by a few days, but no luck.
![Capture-Guj-black](https://github.com/OpenSpace/OpenSpace/assets/6321069/f84d25b1-7da2-4b94-bd16-2c0db72c2306)

**Comment by hn-88** [comment]
I thought perhaps deleting the mrf_cache/Earth folder would cause the data to be re-downloaded. But the problem persisted for Feb 5 2024 (from this recording - https://github.com/hn-88/openspace-scripts/blob/main/user/recordings/zoom%20out%20from%20PSN.osrectxt ). Then, tried for a different date - 23 Feb. Still the same.

**Comment by alexanderbock** [comment]
That is strange.  I can't reproduce it on my end
![image](https://github.com/OpenSpace/OpenSpace/assets/4921724/35e5ea07-72d4-4390-b195-d2d8613c1ae8)

Could you try checking in which layer the error is?  If you go to the Scene -> Current Focus: Earth -> Renderable -> Layers -> Color Layers menu and disable _ESRI VIIRS Combo_ and then first toggle _ESRI World Imagery_  and then _VIIRS SNPP (Temporal)_, the black bar should appear in exactly one of those two layers
![Uploading image.png…]()

**Comment by hn-88** [comment]
Thanks @alexanderbock - I couldn't reproduce the issue either, on another computer. Thanks for the tip on which layers to disable to check. After I find the problematic layer, I suppose I can just delete those files, it would automatically re-download, and all would be well? Will report back tomorrow.

**Comment by alexanderbock** [comment]
Yea, remove the mrf_cache as you did should have done that.  Would be great to try that again and see if it helps.  My best guess is that there was a spurious download error that got saved in your cache

**Comment by hn-88** [comment]
> Could you try checking in which layer the error is? If you go to the Scene -> Current Focus: Earth -> Renderable -> Layers -> Color Layers menu and disable ESRI VIIRS Combo and then first toggle ESRI World Imagery and then VIIRS SNPP (Temporal), the black bar should appear in exactly one of those two layers

I did this. First, I did this without deleting anything in mrf_cache. I went to 2000 km over Gujarat, saw the black bar, then disabled ESRI VIIRS Combo, entire globe became blue as expected. Turning on ESRI World Imagery --> No black bar. Turned it off. Then turned on VIIRS SNPP (Temporal) --> Again no black bar. This was with the time set to the default "yesterday", that is 24 Feb 2024 03:45 GMT. 

Then I tried 05 Feb 2024 04:10 GMT with each of those layers. Again no black bar. All good so far.

Then I tried playing back my osrectxt file, which has the time paused at 05 Feb 2024 05:40 GMT.
With VIIRS SNPP (Temporal) only enabled, there were some tiling artifacts at other places at different altitudes - 400 and 800 km. But the black bar at 2000 km was gone. Screenshots (will add on editing this comment, from the other computer)

With ESRI World imagery only selected, no more black bar at 2000 km. No other artifacts, either.

Then I turned off ESRI World Imagery, and turned on ESRI VIIRS combo. Now the black bar reappeared. So, it's in this layer that the artifact is present. Screenshot: 

Then I clicked Reset for this layer. Still the bar remained. Screenshot: (will add on edit)


Now I disabled the layer, THEN clicked Reset. Then enabled this layer. Again the bar appeared. I had a network monitor open, and my network is pretty slow. It did not re-download. So, Reset did not delete the local cache without restarting OpenSpace.

Now I disabled ESRI VIIRS combo, clicked Reset, then exited OpenSpace and opened it again with the default profile, so with ESRI VIIRS combo enabled. But again the black bar appears. 

So, if you tell me which files to delete manually, I can try that.

**Comment by hn-88** [comment]
OK, I now tried deleting ALL the contents of C:\OpenSpace-0.19.2\mrf_cache\Earth

Still the black bar is seen.
![Captureafterdeletingmrf_cache](https://github.com/OpenSpace/OpenSpace/assets/6321069/ad14a19b-10c9-4c49-a3f9-2fac069eb6b4)

Then I thought I should perhaps try without a VPN - all the trials above were done using ProtonVPN to speed up downloading of zip files which our ISP throttles. Without the VPN also, again tried deleting all the contents of mrf_cache\Earth, tried again, still the black bar appears.

![Capturewithoutvpn](https://github.com/OpenSpace/OpenSpace/assets/6321069/6d87632e-386e-47f8-a5ba-ba55af221ba5)

Maybe this issue is not in the mrf_cache, maybe it is in the sync directory or somewhere?

**Comment by alexanderbock** [comment]
None of the tiling is in stored in the sync folder.   Maybe a dumb question but have you changed the directory where the MRF cache is located?  There is a setting in the openspace.cfg file called `MRFCacheLocation` that controls where the cache ends up with.  Based on your description, it _has_ to be the MRF cache on your machine.  It can't be some HTTP cache upstream because that would corrupt both the combo layer and the individual one.  The only thing the combo layer does is set a level at which we switch between the two provided layers (the ones you tested individually).

Otherwise I really can't explain how this could happen.  Maybe one last test:
1. In the "Color Layers" disable "Blend between levels"
2. In "Overlays" disable "Blend between levels"
3. In "Overlays" enable the "Tile Indices" layer
4. In "Debug", enable "Show chunk edges"

When you then navigate to see the black back and zoom in and out you'll be able to see the tile in which the bar is. Judging on the picture, it looks like `{ x:44, y:11, level: 6}`.   With that information you can manually look at the tile in a browser by going to:
`https://wi.maptiles.arcgis.com/arcgis/rest/services/World_Imagery/MapServer/tile/${z}/${y}/${x}.jpg`  while replacing the z with level-1, and the y and x values; so in this case:

https://wi.maptiles.arcgis.com/arcgis/rest/services/World_Imagery/MapServer/tile/5/11/44.jpg

and then see if the black bar is in there as well.  I assume that it is not

**Comment by alexanderbock** [comment]
Can you also try loading with the MRF cache disabled, just to ensure that it is not some HTTP cache upstream?

**Comment by hn-88** [comment]
> None of the tiling is in stored in the sync folder. Maybe a dumb question but have you changed the directory where the MRF cache is located? There is a setting in the openspace.cfg file called `MRFCacheLocation` that controls where the cache ends up with. 

Checked C:\OpenSpace-0.19.2\openspace.cfg --> MRFCacheLocation = "${BASE}/mrf_cache" - I had not changed it.

>Based on your description, it _has_ to be the MRF cache on your machine. It can't be some HTTP cache upstream because that would corrupt both the combo layer and the individual one. The only thing the combo layer does is set a level at which we switch between the two provided layers (the ones you tested individually).
> 
> Otherwise I really can't explain how this could happen. Maybe one last test:
> 
> 1. In the "Color Layers" disable "Blend between levels"
> 2. In "Overlays" disable "Blend between levels"
> 3. In "Overlays" enable the "Tile Indices" layer
> 4. In "Debug", enable "Show chunk edges"
> 

I don't see a Debug option. So, I just performed steps 1,2,3. Again, with only ESRI selected, no bar. With only VIIR selected, no bar. (But I see that the VIIRS layer shows a lot of clouds over the area which are not seen in the combo layer, so maybe there is some partial data which the combo layer uses.) With the combo layer only selected, since the blending is off, the black bar appears and disappears more abruptly instead of fading in and out, and appears at a slightly higher altitude, screenshot below.

![Captureblending off](https://github.com/OpenSpace/OpenSpace/assets/6321069/cb6b94fa-78e3-457d-8aed-47a44c6e5723)


> When you then navigate to see the black back and zoom in and out you'll be able to see the tile in which the bar is. Judging on the picture, it looks like `{ x:44, y:11, level: 6}`. With that information you can manually look at the tile in a browser by going to: `https://wi.maptiles.arcgis.com/arcgis/rest/services/World_Imagery/MapServer/tile/${z}/${y}/${x}.jpg` while replacing the z with level-1, and the y and x values; so in this case:
> 
> https://wi.maptiles.arcgis.com/arcgis/rest/services/World_Imagery/MapServer/tile/5/11/44.jpg
> 
> and then see if the black bar is in there as well. I assume that it is not

I tried that link. No bar. Also, no bar when only ESRI World Imagery is selected, so that would be expected. 

If the combo only adds ESRI and VIIRS, I'm guessing that it actually adds some "cloud-free" VIIRS day or something like that and not the VIIRS temporal layer itself (which has lots of clouds seen). And I suspect that the tiling issue might be with that particular "cloud-free" VIIRS data, which might be being cached upstream since this computer is behind a IDS/ caching firewall.

**Comment by hn-88** [comment]
> Can you also try loading with the MRF cache disabled, just to ensure that it is not some HTTP cache upstream?

Aha! Problem found. I thought I would delete the entire MRF cache and then disable it. Going to the mrf_cache directory, I found an "unspecified" directory, not under the "Earth" directory. I deleted that directory, and hey presto! no more black bar! Also, the clouds are seen in the combo layer now, which were not seen earlier as I noted in my previous comment.

![image](https://github.com/OpenSpace/OpenSpace/assets/6321069/69a42fea-13bd-4426-b6d2-4df258422b6d)
